File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    bn1 = bin(n)[2:] #bin number
    bn1 = list(bn1)
    count = 0
    result = ''
    for i in range(0, len(bn1)): #n의 1 갯수
        if(bn1[i] == '1'):
            count += 1
    
    for i in range(n+1, 1000000):
        count2 = 0
        bn2 = bin(i)[2:]
        bn2 = list(bn2)
        for j in range(0, len(bn2)):
            if(bn2[j] == '1'):
                count2 += 1
        
        if(count == count2):
            result = ''.join(bn2)
            break;
    result = '0b' + result
    
    return int(result, 2)

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.js
function solution(n) {
    bn1 = n.toString(2); // n을 이진수 문자열로 변환
    count = 0;
    for (i = 0; i < bn1.length; i++) { // n의 1의 개수 세기
        if (bn1[i] == '1') {
            count++;
        }
    }

    for (i = n + 1; i < 1000000; i++) {
        count2 = 0;
        bn2 = i.toString(2); // i를 이진수 문자열로 변환
        for (j = 0; j < bn2.length; j++) {
            if (bn2[j] == '1') {
                count2++;
            }
        }

        if (count == count2) {
            return parseInt(bn2, 2); // 이진수 문자열을 정수로 변환하여 반환
        }
    }
}
